presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java

```diff
@@ -70,11 +70,28 @@ protected void validateRootGroups(ManagerSpec managerSpec)
     {
         ImmutableList.Builder<ResourceGroupSelector> selectors = ImmutableList.builder();
         for (SelectorSpec spec : managerSpec.getSelectors()) {
+            validateSelectors(managerSpec.getRootGroups(), spec.getGroup().getSegments());
             selectors.add(new StaticSelector(spec.getUserRegex(), spec.getSourceRegex(), spec.getGroup()));
         }
         return selectors.build();
     }
 
+    private void validateSelectors(List<ResourceGroupSpec> groups, List<ResourceGroupNameTemplate> selectorGroups)
+    {
+        if (selectorGroups.isEmpty()) {
+            return;
+        }
+        ResourceGroupNameTemplate groupName = selectorGroups.get(0);
+        Optional<ResourceGroupSpec> match = groups
+                .stream()
+                .filter(groupSpec -> groupSpec.getName().equals(groupName))
+                .findFirst();
+        if (!match.isPresent()) {
+            throw new IllegalArgumentException(format("Selector refers to nonexistent group: %s", groupName));
```

Comment:
I think this message is going to be hard to understand. I would change this to use an iterative approach instead of recursion, so that you can print the fully qualified group name. The test you have below has an exception for the group "X", but there could be two groups with that name in different parts of the tree, like "a.b.X" and "a.c.X"